### PR TITLE
Add support for transclusion

### PR DESCRIPTION
**Pull request by gregbty**
This should add the last fix for #36.

### CODE REVIEW

**Comment by gregbty** [comment]
This does not support transclusion of Angular directives, only normal HTML elements and other React components.

File: package.json [hunk 1]
```diff
@@ -74,6 +75,9 @@
     "@types/angular": ">=1.5",
     "@types/lodash.frompairs": "^4.0.3",
     "angular": ">=1.5",
+    "html-react-parser": "^0.4.3",
+    "htmltojsx": "^0.3.0",
+    "install-peers": "^1.0.2",
```
Comment (bcherny) [inline package.json]: Please rm `install-peers`.

File: README.md [hunk 2]
```diff
@@ -59,6 +59,20 @@ If `propTypes` are defined and you passed in a 2nd argument, the argument will o
 ></my-component>
 ```
 
+You can also transclude HTML content which will be rendered as children of the React component.
```
Comment (bcherny) [inline README.md]: Please add a note about Angular and React too.

**Comment by sdvg** [comment]
I really like this feature, but I discovered two problems when trying your branch:

1) It doesn't work when there are text nodes in transcluded content, so even when using a HTML tag, any surrounding whitespace will cause an exception. It's this line: https://github.com/coatue-oss/react2angular/pull/76/files#diff-ce8d93a16385590b555954a5f69400caR64


2) When using webpack I get this compile error:

```
ERROR in ./node_modules/xmlhttprequest/lib/XMLHttpRequest.js
Module not found: Error: Can't resolve 'child_process' in '<...>/node_modules/xmlhttprequest/lib'
 @ ./node_modules/xmlhttprequest/lib/XMLHttpRequest.js 15:12-36
 @ ./node_modules/jsdom-no-contextify/lib/jsdom/browser/Window.js
 @ ./node_modules/jsdom-no-contextify/lib/jsdom/browser/index.js
 @ ./node_modules/jsdom-no-contextify/lib/jsdom.js
 @ ./node_modules/htmltojsx/src/htmltojsx.js
 @ ./node_modules/react2angular/index.js
 @ ./src/app/react-components/index.js
 @ ./src/index.js
 @ multi ./src/index.js webpack-hot-middleware/client?reload=true
```

It's caused by jsdom which tries to call Node methods when `IN_BROWSER` is not defined. I fixed it by adding the following to my webpack config:

```
new webpack.DefinePlugin({
  IN_BROWSER: true,
})
```

Not sure if there is a better solution to this, but it might be worth documenting.

**Comment by gregbty** [comment]
I'll look at trying to resolve this later this week. I did forget to mention that I was unable to get transclusion to work properly for non-React components (Angular components or HTML nodes), but I'll  take a look at that again as well.

**Comment by Dozalex** [comment]
@gregbty Any good news?

**Comment by agatac** [comment]
Hi, what's the status on this one?

**Comment by coreysnyder** [comment]
I'm also interested in this one. I've began using `react2angular` recently to try and prototype a new idea and ran into this issue almost immediately when trying to nest 2 angular components inside of my react component with `this.props.children'

**Comment by gregbty** [comment]
I ran into problems binding raw HTML nodes within React within Angular templates and also transclusion of Angular components within React (react2angular-ified Angular components). I do not think either are possible without some extreme hacks. Merging this PR without those two features may end up causing more confusion than its worth. We solved this problem by managing those nested components directly in React.

**Comment by gregbty** [comment]
@gairon see https://github.com/coatue-oss/react2angular/pull/76#issuecomment-403858928. I'm going to close to this PR since it is no longer actively being worked on.

**Pull request closed without merging**